torch/autograd/gradcheck.py

```diff
@@ -302,16 +320,45 @@ def fn(input):
         analytical, reentrant, correct_grad_sizes = get_analytical_jacobian(tupled_inputs, o, nondet_tol=nondet_tol)
         numerical = get_numerical_jacobian(fn, tupled_inputs, eps=eps)
 
-        if not correct_grad_sizes:
+        out_is_complex = o.is_complex()
+
+        if out_is_complex:
+            # analytical vjp with v = 1.0j
+            analytical_with_imag_v, reentrant_with_imag_v, correct_grad_sizes_ \
+                = get_analytical_jacobian(tupled_inputs, o, nondet_tol=nondet_tol, v=1j)
+
+        if not correct_grad_sizes or (out_is_complex and not correct_grad_sizes_):
```

Comment:
It would be nice to update the error message to tell which of the two does not have the correct size?